src/cryptsetup/cryptsetup-generator.c

```diff
@@ -181,6 +181,48 @@ static int generate_keydev_mount(
         return 0;
 }
 
+static int generate_keydev_umount(const char *name,
+                                  const char *keydev_mount,
+                                  char **umount_unit) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *u = NULL, *name_escaped = NULL, *mount = NULL;
+        int r;
+
+        assert(name);
+        assert(umount_unit);
+
+        name_escaped = cescape(name);
+        if (!name_escaped)
+                return -ENOMEM;
+
+        u = strjoin("keydev-", name_escaped, "-umount.service");
+        if (!u)
+                return -ENOMEM;
+
+        r = unit_name_from_path(keydev_mount, ".mount", &mount);
+        if (r < 0)
+                return r;
+
+        r = generator_open_unit_file(arg_dest, NULL, u, &f);
+        if (r < 0)
+                return r;
+
+        fprintf(f,
+                "[Unit]\n"
+                "DefaultDependencies=no\n"
+                "After=%s\n\n"
+                "[Service]\n"
+                "Type=exec\n"
+                "ExecStart=-" UMOUNT_PATH " %s\n\n", mount, keydev_mount);
```

Comment:
shouldn't this have RemainAfterExit= so that it is not pulled in again?

Reply:
I actually want the opposite.

After switchroot keydev mount gets pulled in, systemd-cryptsetup doesn't do anything because it is RemainAfterExit=yes and I want the umount service to also get pulled in so that it umounts the keydev mount and that doesn't remain lingering.